tests/unit/lms/services/d2l_api/client_test.py

```diff
@@ -118,23 +130,135 @@ def test_group_set_groups_with_user_id(self, svc, d2l_groups_schema, groups):
+                            "ModuleId": 1,
+                            "LastModifiedDate": "DATE 1",
+                            "Title": "MODULE 1",
+                            "Topics": [
+                                {
+                                    "Identifier": "FILE 1",
+                                    "TypeIdentifier": "File",
+                                    "Title": "TITLE 1",
+                                    "LastModifiedDate": "DATE 1",
+                                }
+                            ],
+                            "Modules": [
+                                {
+                                    "ModuleId": 2,
+                                    "LastModifiedDate": "DATE 2",
+                                    "Title": "MODULE 2",
+                                    "Topics": [
+                                        {
+                                            "Identifier": "FILE 2",
+                                            "TypeIdentifier": "File",
+                                            "Title": "TITLE 2",
+                                            "LastModifiedDate": "DATE 2",
+                                        }
+                                    ],
+                                }
+                            ],
+                        }
+                    ]
+                },
+                [
+                    {
+                        "display_name": "MODULE 1",
+                        "id": 1,
+                        "updated_at": "DATE 1",
+                        "type": "Folder",
+                        "children": [
+                            {
+                                "display_name": "TITLE 1",
+                                "id": "FILE 1",
+                                "type": "File",
+                                "updated_at": "DATE 1",
+                            },
+                            {
+                                "display_name": "MODULE 2",
+                                "id": 2,
+                                "type": "Folder",
+                                "updated_at": "DATE 2",
+                                "children": [
+                                    {
+                                        "display_name": "TITLE 2",
+                                        "id": "FILE 2",
+                                        "type": "File",
+                                        "updated_at": "DATE 2",
+                                    },
+                                ],
+                            },
+                        ],
+                    },
+                ],
+            ),
+        ],
+    )
+    def test_list_files(
         self,
         svc,
-        D2LTableOfContentsSchema,
         basic_client,
+        modules,
+        files,
     ):
-        svc.course_table_of_contents("COURSE_ID")
+        basic_client.request.return_value.json.return_value = modules
```

Comment:
This test is not mocking the schema.

I find this a more useful test as we can test what we'll return given a response from the API direclty.